e2e_tests/tests/experiment/test_custom_searcher.py

```diff
@@ -338,6 +340,9 @@ def test_run_asha_batches_exp(tmp_path: pathlib.Path) -> None:
     for trial in response_trials:
         assert trial.state == bindings.determinedexperimentv1State.STATE_COMPLETED
 
+    # clean up
+    client._determined = None
```

Comment:
This line will be skipped if _anything_ in the test raises an exception. Here is my proposal:

```
@pytest.fixture
def client_login() -> Generator[None, None, None]:
    client.login(master=conf.make_master_url())
    yield
    client._determined = None

...
@pytest.mark.nightly
def test_run_asha_batches_exp(tmp_path: pathlib.Path, client_login: None) -> None:
```

Then you don't have to do login and cleanup inside the test, just add `client_login` as an argument.